llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -1279,9 +1278,6 @@ bool AddressSanitizer::isInterestingAlloca(const AllocaInst &AI) {
       (AI.getAllocatedType()->isSized() &&
```

Comment:
isInterestingAlloca contains other checks. There is no point to have a flag to enable instrumentation of allocas which we can't instrument.

Reply:
ClSkipPromotableAllocas on line 1315 is confusing, and maybe unnecessary?
but after your rename it's still confusing

this check applies only to accesses, there is other stuff done to instrument alloca, and ClSkipPromotableAllocas was applied there thru isInterestingAlloca

Reply:
In `ClInstrumentUninterestingAllocas=0 mode`, `!(SSGI && SSGI->isSafe(AI))` can ignore some `AllocaInst`s that `SSGI->stackAccessIsSafe(*Inst)` will return false.

My original thought is that `ClInstrumentUninterestingAllocas` isn't useful and we should just remove it.

If you think `ClInstrumentUninterestingAllocas` is unnecessary, I will remove it.

Reply:
Yes, either keep as ClSkipPromotableAllocas=false, just in case, or remove it.
Something similar but not the same is not needed.